src/main/kotlin/systems/ajax/malov/stockanalyzer/repository/impl/MongoStockRecordRepository.kt

```diff
@@ -0,0 +1,163 @@
+package systems.ajax.malov.stockanalyzer.repository.impl
+
+import com.mongodb.client.MongoCollection
+import com.mongodb.client.model.Filters.gte
+import com.mongodb.client.model.Projections.excludeId
+import com.mongodb.client.model.Projections.fields
+import com.mongodb.client.model.Projections.include
+import com.mongodb.client.model.Sorts
+import org.bson.Document
+import org.bson.types.Decimal128
+import org.springframework.data.domain.Sort
+import org.springframework.data.mongodb.core.MongoTemplate
+import org.springframework.data.mongodb.core.aggregation.AccumulatorOperators.Avg
+import org.springframework.data.mongodb.core.aggregation.Aggregation
+import org.springframework.data.mongodb.core.aggregation.AggregationExpression
+import org.springframework.data.mongodb.core.aggregation.GroupOperation
+import org.springframework.data.mongodb.core.aggregation.MatchOperation
+import org.springframework.data.mongodb.core.aggregation.ProjectionOperation
+import org.springframework.data.mongodb.core.query.Criteria
+import org.springframework.stereotype.Repository
+import systems.ajax.malov.stockanalyzer.entity.MongoStockRecord
+import systems.ajax.malov.stockanalyzer.repository.StockRecordRepository
+import java.math.BigDecimal
+import java.time.Instant
+import java.time.temporal.ChronoUnit
+import java.util.Date
+import org.springframework.data.mongodb.core.mapping.Document as SDocument
+
+@Repository
+class MongoStockRecordRepository(
+    private val mongoTemplate: MongoTemplate,
+) : StockRecordRepository {
+    override fun insertAll(mongoStockRecords: List<MongoStockRecord>): List<MongoStockRecord> {
+        return mongoTemplate.insertAll(mongoStockRecords).toList()
+    }
+
+    override fun findAllStockSymbols(): List<String> {
+        return mongoTemplate.getCollection(MongoStockRecord.COLLECTION_NAME)
+            .distinct(MongoStockRecord::symbol.name, String::class.java)
+            .toList()
+    }
+
+    override fun findTopNStockSymbolsWithStockRecords(n: Int): Map<String, List<MongoStockRecord>> {
```

Comment:
you can accept here 'from' and 'to' dates to not use Instant.now in the repository. and then your repository tests may become independent from each other